proposals/TP1-xds-transport-next.md

```diff
@@ -0,0 +1,864 @@
+configuration from some canonical xDS server.
+
+Client A bootstrap:
+
+
+```
+dynamic_resources:
+  lds_resources_locator:
+    name: xdstp://some-authority/envoy.config.listeners.v3.Listener/a-listeners/*
+  lds_config:
+    authorities:
+    - name: some-authority
+  … rest of ConfigSource pointing at xDS relay proxy … 
+```
+
+
+Client A LDS `DeltaDiscoveryRequest` sent to xDS relay proxy:
+
+
+```
+resource_names_subscribe:
+- xdstp://some-authority/envoy.config.listeners.v3.Listener/a-listeners/*
+```
+
+Client B bootstrap:
+
+
+```
+dynamic_resources:
+  lds_resources_locator:
+    name: xdstp://some-authority/envoy.config.listeners.v3.Listener/b-listeners/*
+  lds_config:
+    authorities:
+    - mame: some-authority
+  … rest of ConfigSource pointing at xDS relay proxy … 
+```
+
+Client B LDS `DiscoveryRequest` sent to xDS relay proxy:
+
+
+```
+resource_names_subscribe:
+- xdstp://some-authority/envoy.config.listeners.v3.Listener/b-listeners/*
+```
+
+
+xDS relay proxy `DeltaDiscoveryRequest` sent to some `some-authority`:
+
+
+```
+resources_names_subscribe:
+- xdstp://some-authority/envoy.config.listeners.v3.Listener/a-listeners/*
+- xdstp://some-authority/envoy.config.listeners.v3.Listener/b-listeners/*
+```
+
+
+The origin for `some-authority` replies with a `DeltaDiscoveryResponse` sent to
+xDS relay proxy:
+
+
+```
+resources:
+- version: 1
+  name: xdstp://some-authority/envoy.config.listeners.v3.Listener/a-listeners/foo
+  resource:
+    "@type": type.googleapis.com/envoy.config.listeners.v3.Listener
+    … foo's Listener payload … 
+- version: 42
+  name: xdstp://some-authority/envoy.config.listeners.v3.Listener/a-listeners/bar
+  resource:
+    "@type": type.googleapis.com/envoy.config.listeners.v3.Listener
+    … bar's Listener payload … 
+- version: 13
+  name: xdstp://some-authority/envoy.config.listeners.v3.Listener/b-listeners/baz
+  resource:
+    "@type": type.googleapis.com/envoy.config.listeners.v3.Listener
+    … baz's Listener payload … 
+```
+
+xDS relay proxy `DeltaDiscoveryResponse` sent to client A:
+
+```
+resources:
+- version: 1
+  name: xdstp://some-authority/envoy.config.listeners.v3.Listener/a-listeners/foo
+  resource:
+    "@type": type.googleapis.com/envoy.config.listeners.v3.Listener
+    … foo's Listener payload … 
+- version: 42
+  name: xdstp://some-authority/envoy.config.listeners.v3.Listener/a-listeners/bar
+  resource:
+    "@type": type.googleapis.com/envoy.config.listeners.v3.Listener
+    … bar's Listener payload … 
+```
+
+xDS relay proxy `DeltaDiscoveryResponse` sent to client B:
+
+```
+resources:
+- version: 13
+  name: xdstp://some-authority/envoy.config.listeners.v3.Listener/b-listeners/baz
+  resource:
+    "@type": type.googleapis.com/envoy.config.listeners.v3.Listener
+    … baz's Listener payload … 
+```
+
+## Rationale
+
+[A discussion of alternate approaches and the trade offs, advantages, and disadvantages of the specified approach.]
```

Comment:
We should probably either fill in these last few sections or else replace the contents with "N/A". :)